MinPlatformPkg/Test/TestPointStubDxe/TestPointStubDxeStandaloneMm.c

```diff
@@ -0,0 +1,341 @@
+/** @file
+
+  Copyright (c) 2017, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <PiDxe.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/DebugLib.h>
+#include <Library/UefiLib.h>
+#include <Library/HobLib.h>
+#include <Library/TestPointLib.h>
+#include <Protocol/AdapterInformation.h>
+#include <Protocol/MmCommunication.h>
+#include <Guid/PiSmmCommunicationRegionTable.h>
+
+UINTN  mSmmTestPointDatabaseSize;
+VOID   *mSmmTestPointDatabase;
+
+VOID
+PublishPeiTestPoint (
+  VOID
+  )
+{
+  EFI_PEI_HOB_POINTERS              Hob;
+  ADAPTER_INFO_PLATFORM_TEST_POINT  *TestPoint;
+  UINTN                             TestPointSize;
+
+  DEBUG ((DEBUG_INFO, "PublishPeiTestPoint\n"));
+
+  Hob.Raw = GetHobList ();
+  while (TRUE) {
+    Hob.Raw = GetNextGuidHob (&gAdapterInfoPlatformTestPointGuid, Hob.Raw);
+    if (Hob.Raw == NULL) {
+      return ;
+    }
+    TestPoint = GET_GUID_HOB_DATA (Hob);
+    TestPointSize = GET_GUID_HOB_DATA_SIZE (Hob);
+
+    TestPointLibSetTable (TestPoint, TestPointSize);
+
+    Hob.Raw = GET_NEXT_HOB (Hob);
+    if (Hob.Raw == NULL) {
+      return ;
+    }
+  }
+}
+
+VOID
+TestPointStubForPei (
+  VOID
+  )
+{
+  PublishPeiTestPoint ();
+}
+
+VOID
+GetTestPointDataStandaloneMm (
```

Comment:
What is the difference between this and the SMM version? I tried to compare the 2 instances, but did not notice anything obvious.

It might be a good idea to reuse the code and just implement/abstract the difference between the 2 MM versions.

Reply:
Yeah, I've left this feedback in a number of places. This one looks fairly trivial to consolidate into a single implementation.

Please look for this throughout all of the changes in the PR and it will greatly reduce the amount of change being introduced to what is functionally required.